### PR TITLE
[dtest] Add a method for Prom remote writes

### PR DESCRIPTION
**Pull request by vpranckaitis**
**What this PR does / why we need it**:

Add a method for Prom remote writes to dtest docker harness 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: src/cmd/tools/dtest/docker/harness/resources/coordinator.go [hunk 1]
```diff
@@ -323,6 +335,77 @@ func (c *coordinator) WriteCarbon(
...
+		url = c.resource.getURL(7201, "api/v1/prom/remote/write")
+
+		reqLabels  = make([]prompb.Label, 0)
+		reqSamples = make([]prompb.Sample, 0)
+	)
+
+	reqLabels = append(reqLabels, prompb.Label{
+		Name:  []byte(model.MetricNameLabel),
+		Value: []byte(name),
+	})
```
Comment (linasm) [inline src/cmd/tools/dtest/docker/harness/resources/coordinator.go:353]: Micro nit (slightly less syntax for the eyes to parse):
```suggestion
		reqLabels = []prompb.Label{{Name: []byte(model.MetricNameLabel), Value: []byte(name)}}
		reqSamples  []prompb.Sample
```
or something in that direction.

File: src/cmd/tools/dtest/docker/harness/resources/coordinator.go [hunk 2]
```diff
@@ -57,6 +61,12 @@ var (
 // ResponseVerifier is a function that checks if the query response is valid.
 type ResponseVerifier func(int, map[string][]string, string, error) error
 
+// Sample is a struct containing a metric value at a given timestamp.
+type Sample struct {
+	Timestamp int64
+	Value     float64
+}
```
Comment (linasm) [inline src/cmd/tools/dtest/docker/harness/resources/coordinator.go:68]: I think you could reuse `ts.Datapoint` (or some other readily available struct).

**Review by linasm — APPROVED** [review]
LGTM with some suggestions.

File: src/cmd/tools/dtest/docker/harness/resources/coordinator.go (hunk 1 above)
Comment (vpranckaitis) [reply src/cmd/tools/dtest/docker/harness/resources/coordinator.go]: https://github.com/m3db/m3/pull/3065/commits/de8c17f6e067c6426c7e18d7b620c8b8967a62d4#diff-36f3f4899036050c43cca0b378f19429cc8ceb118ebffd0c42f672ee13d507e8L344-L353

File: src/cmd/tools/dtest/docker/harness/resources/coordinator.go (hunk 2 above)
Comment (vpranckaitis) [reply src/cmd/tools/dtest/docker/harness/resources/coordinator.go]: https://github.com/m3db/m3/pull/3065/commits/de8c17f6e067c6426c7e18d7b620c8b8967a62d4#diff-36f3f4899036050c43cca0b378f19429cc8ceb118ebffd0c42f672ee13d507e8R74

**Pull request merged**